File: cpp/cpp_1000-/2399.cc
// C++
// leetcode 2399
// https://leetcode.cn/problems/check-distances-between-same-letters/

class Solution {
public:
    bool checkDistances(string s, vector<int>& distance) {
        bool vis[26];
        memset(vis, 0, sizeof(vis));
        int n = s.size();
        for (int i = 0; i < n; i++) {
        	int u = s[i] - 'a';
        	if (vis[u]) {
        		if (distance[u] != i)
        			return false;
        	} else {
        		vis[u] = true;
        		distance[u] += i + 1;
        	}
        }
        return true;
    }
};

File: Go/Go_1000-/2399.go
// Go
// leetcode 2399
// https://leetcode.cn/problems/check-distances-between-same-letters/

func checkDistances(s string, distance []int) bool {
	vis := make([]bool, 26)
	n := len(s)
	for i, r := range s {
		u := int(r - 'a')
		if vis[u] {
			continue
		}
		if i+distance[u]+1 >= n || s[i+distance[u]+1] != byte(r) {
			return false
		}
		vis[u] = true
	}
	return true
}